vnext/ReactWindowsCore/WinRTWebSocketResource.cpp

```diff
@@ -354,6 +391,40 @@ void WinRTWebSocketResource::Synchronize()
   }
 }
 
+string WinRTWebSocketResource::GetRestrictedErrorMessage() noexcept
+{
+  string result;
+  IRestrictedErrorInfo* errorInfo;
+  if (SUCCEEDED(GetRestrictedErrorInfo(&errorInfo)))
+  {
+    BSTR description;
+    HRESULT error;
+    BSTR restrictedDesc;
+    BSTR capabilitySid;
+    if (SUCCEEDED(errorInfo->GetErrorDetails(&description, &error, &restrictedDesc, &capabilitySid)))
+    {
+      result = Utf16ToUtf8(description) + ": " + Utf16ToUtf8(restrictedDesc);
+    }
+    else
+    {
+      result = "Unknown error";
+    }
+    // BSTR's allocated memory must be explicitly released.
+    // See https://docs.microsoft.com/en-us/cpp/atl-mfc-shared/allocating-and-releasing-memory-for-a-bstr
+    SysFreeString(description);
+    SysFreeString(restrictedDesc);
+    SysFreeString(capabilitySid);
+
+    //TODO: Free errorInfo memory?
```

Comment:
@asklar Does `IRestrictedErrorInfo* errorInfo` require explicit release / delete?

Reply:
it's a COM interface, so yes.
Use a smart COM pointer.